packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -83,12 +83,15 @@ function classOrObjectMethod(path: NodePath, callId: Object) {
 
   node.async = false;
 
-  const container = t.functionExpression(null, [], t.blockStatement(body.body), true);
+  const container = t.functionExpression(node.key, [], t.blockStatement(body.body), true);
   container.shadow = true;
   body.body = [
     t.returnStatement(t.callExpression(
-      t.callExpression(callId, [container]),
```

Comment:
By using `apply` instead of calling the function directly, we ensure that the function has the same `this` value as the original method call, so that the error's stack trace picks up the className like it would as if it was running the original async method.